lib/model/rwfolder.go

```diff
@@ -87,6 +89,8 @@ type rwFolder struct {
 	copiers      int
 	pullers      int
 	shortID      uint64
+	newVersion   protocol.Vector
```

Comment:
Here's what I suggest:

- Remove the added curVersion and newVersion attributes
- Change the findNewestUpdateID function to take two version vectors, i.e. `func findNewestUpdateID(old, new protocol.Vector) string`. The check it needs to do is a bit more involved too; the current code assumes the two vectors have the same device IDs in them, but the vectors can be totally different i.e. `{id1: 20, id2: 30}` vs `{id3: 42, id1: 20}`.
- Pass the `cur.Version` and `file.Version` that is compared to see if there *is* a conflict to moveForConflict and then onwards to findNewestUpdateID.